utils/transaction-provider/base-transaction-provider.ts

```diff
@@ -0,0 +1,133 @@
+import { TW, type WalletCore } from "@trustwallet/wallet-core";
+import type { CoinType } from "@trustwallet/wallet-core/dist/src/wallet-core";
+import type { ChainKey } from "~utils/constants";
+import type {
+  SignatureProps,
+  TransactionProps,
+  VaultProps,
+} from "~utils/interfaces";
+import {
+  JsonRpcProvider,
+  Transaction,
+  encodeBase64,
+  formatUnits,
+  hexlify,
+  isHexString,
+  keccak256,
+  randomBytes,
+  toUtf8Bytes,
+  toUtf8String,
+} from "ethers";
+
+import { create, toBinary } from "@bufbuild/protobuf";
+
+
+import {
+  EthereumSpecificSchema,
+  type EthereumSpecific,
+} from "~protos/blockchain_specific_pb";
+import { CoinSchema } from "~protos/coin_pb";
+import {
+  KeysignMessageSchema,
+  KeysignPayloadSchema,
+  type KeysignPayload,
+} from "~protos/keysign_message_pb";
+
+
+
+interface ChainRef {
+  [chainKey: string]: CoinType;
+}
+export abstract class BaseTransactionProvider {
+  protected chainKey: ChainKey;
+  protected chainRef: ChainRef;
+  protected dataEncoder: (data: Uint8Array) => Promise<string>;
+  protected walletCore: WalletCore;
+  protected keysignPayload: KeysignPayload
+  constructor(
+    chainKey: ChainKey,
+    chainRef: ChainRef,
+    dataEncoder: (data: Uint8Array) => Promise<string>,
+    walletCore: WalletCore
+  ) {
+    this.chainKey = chainKey;
+    this.chainRef = chainRef;
+    this.dataEncoder = dataEncoder;
+    this.walletCore = walletCore;
+  }
+
+  protected encryptionKeyHex = (): string => {
+    const keyBytes = randomBytes(32);
+
+    return Array.from(keyBytes)
+      .map((byte) => byte.toString(16).padStart(2, "0"))
+      .join("");
+  };
+
+  protected stripHexPrefix = (hex: string): string => {
+    return hex.startsWith("0x") ? hex.slice(2) : hex;
+  };
+
+  public getPreSignedImageHash = (
+    preSignedInputData: Uint8Array
+  ): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const preHashes = this.walletCore.TransactionCompiler.preImageHashes(
+        this.chainRef[this.chainKey],
+        preSignedInputData
+      );
+
+      const preSigningOutput =
+        TW.TxCompiler.Proto.PreSigningOutput.decode(preHashes);
+      if (preSigningOutput.errorMessage !== "")
+        reject(preSigningOutput.errorMessage);
+
+      const imageHash = this.walletCore.HexCoding.encode(
+        preSigningOutput.dataHash
+      )?.replace(/^0x/, "");
```

Comment:
there is a method `stripHexPrefix` right above , and here we also provide a different way to remove `0x` , can we keep it consistent?